src/lib/components/actions.svelte

```diff
@@ -143,8 +164,10 @@
 			stateCopy.mermaid = JSON.parse(stateCopy.mermaid);
 		}
 		const b64Code = toBase64(JSON.stringify(stateCopy), true);
+		const krokiCode = getKrokiCode(stateCopy.code);
```

Comment:
As this code will be run on every character change in the diagram, what would be the effect on lower end machines as compression is a CPU intensive activity?

Reply:
I would imagine that in reality, the impact would be minimal as (according to pako's README benchmarks), the library performs well and we are usually dealing with a very small amount of data. Having said that, I don't think we should waste any CPU resources if there is no clear benefit for it. I have updated the code in 4472b19 to generate URL (and thus run compression) only on-demand.